core/server/common/src/main/java/alluxio/master/audit/AsyncUserAccessAuditLogWriter.java

```diff
@@ -0,0 +1,79 @@
+package alluxio.master.audit;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public final class AsyncUserAccessAuditLogWriter {
+  private static final int QUEUE_SIZE = 10000;
+  private static final int BLOCKING_TIMEOUT_US = 10;
+  private static final Logger LOG = LoggerFactory.getLogger(AsyncUserAccessAuditLogWriter.class);
+  private final boolean mEnabled;
+  private volatile boolean mStopped;
+  private LinkedBlockingQueue<AuditContext> mAuditLogEntries;
+
+  public AsyncUserAccessAuditLogWriter() {
+    mEnabled = Boolean.parseBoolean(Configuration.get(PropertyKey.MASTER_AUDIT_LOGGING_ENABLED));
+    if (mEnabled) {
+      mAuditLogEntries = new LinkedBlockingQueue<>(QUEUE_SIZE);
+    }
+    mStopped = true;
+  }
+
+  public boolean isEnabled() { return mEnabled; }
```

Comment:
could we make it so that the logger is always enabled, but we only create it if the `MASTER_AUDIT_LOGGING_ENABLED` flag is set?